src/main/java/com/pubnub/api/models/consumer/objects_vsp/space/Space.java

```diff
@@ -0,0 +1,38 @@
+package com.pubnub.api.models.consumer.objects_vsp.space;
+
+import com.pubnub.api.models.consumer.objects_api.PNObject;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.experimental.Accessors;
+
+@Getter
+@Setter
+@Accessors(chain = true)
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+@NoArgsConstructor
+public class Space extends PNObject {
```

Comment:
Shouldn't we have a `SpaceId` as a type for `id` field for this class?

Reply:
Yes. Good catch. I will create class similar to PNObject but with no id and add:
private SpaceId id;   
to Space.java. 
I will also add custom interceptor to map id as String in REST response to SpaceId .

public class SpaceIdDeserializer implements JsonDeserializer<SpaceId> {
    @SneakyThrows
    @Override
    public SpaceId deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        String spaceIdValue = json.getAsString();
        return new SpaceId(spaceIdValue);
    }
}